src/utils/config.ts

```diff
@@ -1,26 +1,29 @@
 import SASjs from '@sasjs/adapter/node'
+import { ServerType, Target } from '@sasjs/utils/types'
 import { readFile, fileExists, folderExists, createFile } from './file'
-import {
-  isAccessTokenExpiring,
-  getNewAccessToken,
-  refreshTokens
-} from './auth-utils'
-import { getVariable } from './utils'
+import { isAccessTokenExpiring, getNewAccessToken, refreshTokens } from './auth'
 import path from 'path'
 import chalk from 'chalk'
 import dotenv from 'dotenv'
+import { Configuration } from '../types'
+import { getConstants } from '../constants'
 
-export async function getConfiguration(pathToFile) {
-  const config = await readFile(pathToFile, false, true).catch((err) => {
-    Promise.resolve(null)
-  })
+/**
+ * Returns an object that represents the SASjs CLI configuration in a given file.
+ * @param {string} pathToFile - the path to the file in question.
+ * @returns {Configuration} configuration object if available.
+ */
+export async function getConfiguration(
+  pathToFile: string
+): Promise<Configuration> {
```

Comment:
this function doesn't return a Promise

Reply:
`async/await` in ES6/TypeScript is based on `Promise`s, so this function _does_ actually return a `Promise`.